resolveImports.mjs

```diff
@@ -1,7 +1,19 @@
+// @ts-check
+
 import { parse } from "es-module-lexer";
 import path from "node:path";
 import { readFile, access } from "node:fs/promises";
 
+/**
+ * Resolved imports.
+ * @typedef {string[] | undefined} ResolvedImports
+ */
```

Comment:
I'm not sure it's a good idea to have this `ResolvedImports` type.

Firstly, if you were to have it as a standalone type, it's better to remove the `| undefined`. `undefined` can't be considered an import. Then functions would return either resolved imports, or else `undefined`: `ResolvedImports | undefined`.

From an API design perspective, it might be better to resolve an empty array instead of the possibility the result is `undefined`.

If you don't include `undefined` in the type, it will make that type more useful in code where it's intending to operate on populated results, instead of having to do something like `NonNullable<ResolvedImports>`.

The name should probably not have `Resolved` in it, because when you name types you describe what that type is, not how that type may have been created, or how that type is intended to be used. For example, a better name might be `ModuleSpecifiers`, or `ModuleUrls` if that's what they are, I'm not really sure.

Secondly, it might be cleaner to just not have a standalone type for `ResolvedImports`. Everywhere just write `Array<string>` and in the JSDoc descriptions explain what the strings are.